lib/rest-adapter.js

```diff
@@ -403,6 +411,11 @@ RestAdapter.prototype._registerMethodRouteHandlers = function(router,
     // Express 4.x only supports delete
     verb = 'delete';
   }
+
+  if (fastRouter.canRegister(route, handler)) {
+    handler.methodName = sharedMethod.name;
+    fastRouter[verb](route, handler);
```

Comment:
If you decide to keep the name on the `handler` function, then please choose a more descriptive property name. The problem with `methodName` is that it's not clear whether it refers to HTTP method or shared/remote method.